state/state.go

```diff
@@ -27,3 +27,41 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+type Key struct {
+	Name string
+	Mode [ModeLen]byte
+}
+
+const (
+	ModeLen = 8
+	Read    = 0
+	Write   = 1
+)
+
+func NewKey(name string, bits ...int) Key {
+	var key Key
+	key.Name = name
+
+	for _, bit := range bits {
+		byteIdx := bit / 8
+		bitIdx := uint(bit % 8)
+
+		if byteIdx < ModeLen {
+			key.Mode[byteIdx] |= (1 << bitIdx)
```

Comment:
This will panic if we use a bit outside of the bound, we should add a bounds check + tests for these new functions

Reply:
Will look into this